zio-crypto/shared/src/main/scala/zio/crypto/random/SecureRandom.scala

```diff
@@ -37,32 +37,34 @@ private final case class SecureRandomLive(randomRef: FiberRef[JSecureRandom]) ex
 
 object SecureRandom {
 
-  val live: Layer[NoSuchAlgorithmException, Has[SecureRandom]] = (for {
-    // Java's SecureRandom can be a major source of lock contention.
-    // Tink wraps Java's SecureRandom in a ThreadLocal to solve this problem.
-    // https://github.com/google/tink/issues/72
-    randomRef <- FiberRef.make[JSecureRandom](new JSecureRandom())
+  val live: Layer[NoSuchAlgorithmException, SecureRandom] = ZLayer {
+    ZIO.scoped {
```

Comment:
Is this the appropriate use of `Scoped`? Wasn't sure for this case, but seems ok.

Reply:
No this will release the scope at the end of the creation of the layer and leak out the released fiber ref.

You need to use `ZLayer.scoped { ... }` which creates a `Scope` for the lifecycle of the layer

Reply:
🤦 Oh yeah. I'll get this updated!